packages/synapse-react-client/src/components/TanStackTable/StyledTanStackTable.tsx

```diff
@@ -1,22 +1,56 @@
-import React from 'react'
-import { flexRender, Table } from '@tanstack/react-table'
+import React, { useMemo } from 'react'
+import { Cell, flexRender, Table } from '@tanstack/react-table'
 import {
   StyledTableContainer,
   StyledTableContainerProps,
 } from '../styled/StyledTableContainer'
+import TableBody from './TableBody'
+import {
+  getColumnSizeCssVariable,
+  getHeaderSizeCssVariable,
+} from './TanStackTableUtils'
 
 type StyledTanStackTableProps<T = unknown> = {
   table: Table<T>
   styledTableContainerProps?: StyledTableContainerProps
+  cellRenderer?: (cell: Cell<T, unknown>) => React.ReactNode
```

Comment:
allow overriding the cell component, allowing Synapse table to pass some custom logic